todo_app/app.py

```diff
@@ -21,39 +21,45 @@ def index():
 def render_cards(id):
     status = False
     cardsList = get_cards(id)
-    print(cardsList)
+    cardsList.sort(key=lambda x: x.status)
+    boardLists = get_boardLists(id)
 
-    return render_template("index.html", cardsList=cardsList, status=status)
+    return render_template(
+        "index.html",
+        cardsList=cardsList,
+        status=status,
+        boardLists=boardLists,
+        boardId=id,
+    )
 
 
-@app.route("/", methods=["POST"])
+@app.route("/add", methods=["POST"])
 def add_new_item():
-    item = request.form["item"]
-    add_item(item)
-    return redirect("/")
+    boardId = request.form.get("board_id")
+    inputItem = request.form.get("inputItem")
+    selectedList = request.form.getlist("selectedList")[0]
+    itemDescription = request.form.get("itemDescription")
+    dueDate = request.form.get("dueDate")
+    date_object = datetime.strptime(dueDate, "%d-%m-%y").date()
```

Comment:
Minor but this errors if a dueDate hasn't been passed - if that's the desired case it would be nice to add a `required` tag to the date input in the HTML to give the user some simple UX to tell them it's necessary

Reply:
Added the required tag